addons/stock/models/stock_production_lot.py

```diff
@@ -11,7 +11,8 @@ class ProductionLot(models.Model):
     _inherit = ['mail.thread','mail.activity.mixin']
     _description = 'Lot/Serial'
     _check_company_auto = True
-
+    _order = 'name'
```

Comment:
to prevent eristic order : use `name,id`

And maybe it make sens to show the last first, `order = 'name desc, id desc'`

note : where there are some search() of `stock.production.lot` in the code, add `order='id'` (name is not indexed).

If it is just an issue in the tree view, you can add `default_oder='name,id' ` in the tree node of the view.